debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/DumpAllOffHeapEntriesCommand.java

```diff
@@ -0,0 +1,120 @@
+import com.ibm.j9ddr.vm29.types.UDATA;
+
+/**
+ * Dump all off-heap entries in the core file.
+ *
+ * usage: !dumpalloffheapentries
+ */
+public class DumpAllOffHeapEntriesCommand extends Command
+{
+	public DumpAllOffHeapEntriesCommand()
+	{
+		addCommand("dumpalloffheapentries", "[help]", "dump all off-heap entries");
+	}
+
+	@Override
+	public void run(String command, String[] args, Context context, PrintStream out) throws DDRInteractiveCommandException
+	{
+		if (0 != args.length) {
+			String argument = args[0];
+
+			if (argument.equalsIgnoreCase("help")) {
+				help(out);
+				return;
+			}
+		}
+
+		try {
+			MM_GCExtensionsPointer extensions = GCBase.getExtensions();
+			MM_SparseVirtualMemoryPointer sparseVirualMemory = extensions.largeObjectVirtualMemory();
+			if (sparseVirualMemory.notNull()) {
+				MM_SparseAddressOrderedFixedSizeDataPoolPointer sparseAddressOrderedFixedSizeDataPool = sparseVirualMemory._sparseDataPool();
+				if (sparseAddressOrderedFixedSizeDataPool.notNull()) {
+					J9HashTablePointer objectToSparseDataTable = sparseAddressOrderedFixedSizeDataPool._objectToSparseDataTable();
+					if (objectToSparseDataTable.notNull()) {
+						try {
+							HashTable<MM_SparseDataTableEntryPointer> readObjectToSparseDataTable = ObjectToSparseDataHashTable.fromJ9HashTable(objectToSparseDataTable);
+							SlotIterator<MM_SparseDataTableEntryPointer> readSlotIterator = readObjectToSparseDataTable.iterator();
+							long count = readObjectToSparseDataTable.getCount();
+							out.printf("List off-heap entries(%,d entries)%n", count);
+							if (0 < count) {
+								out.printf("+------------------+------------------+------------------%n");
+								out.printf("|  array object    |  data address    |      size        %n");
+								out.printf("+------------------+------------------+------------------%n");
+								String outputFormat = " 0x%016x 0x%016x 0x%016x%n";
+
+								while (readSlotIterator.hasNext()) {
+									MM_SparseDataTableEntryPointer readSparseDataTableEntryPtr2 = MM_SparseDataTableEntryPointer.cast(readSlotIterator.nextAddress());
+									VoidPointer dataPrt = readSparseDataTableEntryPtr2._dataPtr();
+									VoidPointer proxyObjPtr = readSparseDataTableEntryPtr2._proxyObjPtr();
+									UDATA size = readSparseDataTableEntryPtr2._size();
+
+									out.append(String.format(
+											outputFormat,
+											proxyObjPtr.getAddress(),
+											dataPrt.getAddress(),
+											size.longValue()));
+								}
+								out.printf("+------------------+------------------+------------------%n");
+							}
+						} catch (CorruptDataException e) {
+							throw new DDRInteractiveCommandException(e);
+						}
+					} else {
+						throw new CorruptDataException("no objectToSparseDataTable");
+					}
+				} else {
+					throw new CorruptDataException("no sparseAddressOrderedFixedSizeDataPool");
+				}
+			} else {
+				throw new CorruptDataException("no sparseVirualMemory");
```

Comment:
Throwing exceptions is not an appropriate control-flow strategy.

The message on line 112 also is not accurate: there may be no off-heap entries even if both `-Xgcpolicy:balanced` and `-XXgc:enableVirtualLargeObjectHeap` are specified.

Reply:
updated, if there is the case no off-heap entries even if both -Xgcpolicy:balanced and -XXgc:enableVirtualLargeObjectHeap are specified,  0 entries would be printed.
```
> !dumpalloffheapentries
List off-heap entries(0 entries)
```

Reply:
When I tested earlier I got nothing:
```
> !dumpalloffheapentries
>
```
now I see:
```
> !dumpalloffheapentries
List off-heap entries(0 entries)
> 
```